src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -637,6 +637,12 @@ private function registerWorkflowConfiguration(array $config, ContainerBuilder $
                 foreach ($workflow['marking_store']['arguments'] as $argument) {
                     $markingStoreDefinition->addArgument($argument);
                 }
+                if ('method' === $workflow['marking_store']['type']) {
+                    $markingStoreDefinition->setArguments([
+                        'state_machine' === $type, //single state
```

Comment:
This assumption is wrong. You can pre-configure with this code, but The value should be overrideable by the enduser.
It **is** possible to have a workflow with a "single state" marking store

Reply:
Actually, this is useless. :+1: with your patch